sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -239,21 +239,23 @@ trait FileSourceScanLike extends DataSourceScanExec {
   // Identifier for the table in the metastore.
   def tableIdentifier: Option[TableIdentifier]
 
-
   lazy val fileConstantMetadataColumns: Seq[AttributeReference] = output.collect {
     // Collect metadata columns to be handled outside of the scan by appending constant columns.
     case FileSourceConstantMetadataAttribute(attr) => attr
   }
 
-  override def vectorTypes: Option[Seq[String]] =
+  protected def sessionState: SessionState = relation.sparkSession.sessionState
```

Comment:
While this approach may resolve the immediate issue, it could incur long-term development costs:
Validating whether newly added code triggers "red highlighting"  in the IDE for every pr submission would be impractical. This might lead to a proliferation of similar follow-up tasks or recurring issues.

Reply:
We can add style check later